### ISSUE TITLE
Feature request: Improved error `async arrow functions are unsupported`

### ISSUE THREAD

**Issue by leotm** [facebook/hermes#1395]
## Problem

async arrow functions (ignored by Babel) trigger Hermes error

https://github.com/facebook/hermes/blob/4ab9f178caf6f0d47df0e0778bec5d1bf394f5c6/lib/IRGen/ESTreeIRGen-func.cpp#L215

causing Metro release bundling to fail `./gradlew :app:installRelease`

## Solution

it would be great if Hermes would output instead

`async arrow functions are unsupported`

## Additional Context

- https://github.com/facebook/hermes/issues/1208
- https://github.com/facebook/hermes/issues/1389

**Comment by tmikov** [comment]
That is a good point. So, basically we need to make the error message more specific. Can you please submit a PR for this?

**Comment by kleydon** [comment]
Has anyone figured out how to configure support for async arrow functions using plug-ins, in the meantime? Would really love to know how to do this.

**Comment by ChALkeR** [comment]
@kleydon yes: https://github.com/ExodusMovement/test/blob/6e928e7267eb5ef19d316e0b09b81d49553bfd8f/bundler/bundle.js#L57-L63 + rest is in esbuild config in the same file

**Comment by fbmal7** [comment]
FWIW, async arrows are implemented in [SH](https://github.com/facebook/hermes/commit/3b1310b3522ef500e9591bdb4c3768aa3f710697?fbclid=IwZXh0bgNhZW0CMTEAAR376Cadl_gNSAB4tHoO0dhrqZnF2Je6_sKXaLh5HIeuewXl1ELzMyRf68M_aem_pnau7wtGaH-qtuMfJOw8sQ)

**Comment by kleydon** [comment]
@ChALkeR , @fbmal7  - thanks!
